index.html

```diff
@@ -78,30 +84,74 @@
         })
       }
 
+      var DOMURL = window.URL || window.webkitURL || window;
+
       // helper functions
+      function loadImageDataIntoCache(imageObject) {
+        var newImg = new Image();
+        newImg.width = imageObject.width;
+        newImg.height = imageObject.height;
+        createImage(newImg, imageObject.url)
+          .then(function(img) {
+            b64ImageDict[imageObject.id] = getB64FromImg(newImg);
+          })
+      }
+
+      function imageFileToB64(src) {
+        return new Promise(function(resolve) {
+          var reader = new FileReader();
+          reader.onload = resolve
+          reader.readAsDataURL(src);
+        })
+      }
+
+      function createImage(img, imageUrl) {
+        return new Promise(function(resolve) {
+          img.crossOrigin = 'anonymous';
+          img.onload = function() {
+            resolve(this)
+          }
+          img.src = imageUrl;
+        });
+      }
+
       function loadImage(src) {
-        if (!src.type.match(/image.*/)) {
-          return;
-        }
-        var reader = new FileReader();
-        reader.onload = function (e) {
-          var imageUrl = e.target.result;
-          var contentType = 'image/png';
-          var newImg = new Image();
-          newImg.crossOrigin = 'anonymous';
-          newImg.onload = function () {
-            app.ports.newImage.send({
+        return imageFileToB64(src)
+          .then(function(e) {
+            b64ImageDict['0'] = e.target.result;
+            var contentType = 'image/png';
+            var blob = b64toBlob(e.target.result.replace('data:image/png;base64,', ''), contentType);
```

Comment:
I do agree about the quality difference between `JPG` and `PNG`, I did also check and we upload images as `PNGs` after we inject them into the editor, so that's all good